src/Substrait.Core/Expression/RelExpression.cs

```diff
@@ -0,0 +1,14 @@
+﻿using Substrait.Relation;
+
+namespace Substrait.Expression
+{
+  /// <summary>
+  /// Base type for all relational expressions, <see cref="Protobuf.Expression"/>
+  /// </summary>
+  abstract public class RelExpression
+  {
+    public virtual void Accept(SubstraitRelVisitor visitor)
```

Comment:
Should this be abstract?  Also, is `SubstraitRelVisitor` intended to visit both relations and expressions?  Or should this be `SubstraitExpressionVisitor`?

Reply:
Good point, separation would be more meaningful. If it is ok with you both, I'd like to fix this in a subsequent PR.